src/routes/api/assets/getById.js

```diff
@@ -1,16 +1,67 @@
 const logger = require("../../../logger");
-const { createSuccessResponse } = require("../../../response");
+const prisma = require("../../../model/data/prismaClient");
+const {
+  createSuccessResponse,
+  createErrorResponse,
+} = require("../../../response");
 
 /**
- * Get an asset by it's id
+ * Get an asset by its id
  */
-
-// eslint-disable-next-line no-unused-vars
 module.exports = async (req, res, next) => {
-  logger.debug(
-    { user: req.user, id: req.params.id },
-    `received request: GET /v1/assets/:assetId`,
-  );
+  try {
+    logger.debug(
+      { user: req.user, id: req.params.assetId },
+      `received request: GET /v1/assets/:assetId`,
+    );
+
+    const assetId = parseInt(req.params.assetId);
+    if (isNaN(assetId)) {
+      return res
+        .status(400)
+        .json(createErrorResponse(400, "Invalid asset ID format"));
+    }
+
+    // Get asset with type-specific data
+    const asset = await prisma.asset.findUnique({
+      where: { id: assetId },
+      include: {
+        character: true,
+      },
+    });
+
+    // Check if asset exists
+    if (!asset) {
+      return res.status(404).json(createErrorResponse(404, "Asset not found"));
```

Comment:
suggestion: We can use `next` instead of `createErrorResponse` to reduce code duplication and improve readability. You can take a look at POST /assets to see how to do this.